src/Motor.Extensions.Hosting.CloudEvents/MotorCloudEvent.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mime;
+using CloudNative.CloudEvents;
+
+namespace Motor.Extensions.Hosting.CloudEvents
+{
+    public static class MotorCloudEventInfo
+    {
+        public static CloudEventsSpecVersion SpecVersion => CloudEventsSpecVersion.V1_0;
+
+        public static readonly IEnumerable<CloudEventAttribute> RequiredAttributes = SpecVersion.RequiredAttributes.Concat(new[]{
+            SpecVersion.TimeAttribute, SpecVersion.DataContentTypeAttribute
+        });
+    }
+
+    public class MotorCloudEvent<TData> where TData : class
+    {
+        private readonly IApplicationNameService _applicationNameService;
+
+        public MotorCloudEvent(
```

Comment:
nit: AFAIK default parameters in constructors are not recommended, personally I'd prefer something like this:

```cs
        public MotorCloudEvent(
            IApplicationNameService applicationNameService,
            TData data,
            Uri source,
            params KeyValuePair<CloudEventAttribute, object>[] extensions) :this(applicationNameService, data, source, null, null, extensions)
        {
            
        }

        public MotorCloudEvent(
            IApplicationNameService applicationNameService,
            TData data,
            Uri source,
            string? id,
            DateTimeOffset? time,
            params KeyValuePair<CloudEventAttribute, object>[] extensions)
        {
            BaseEvent = new CloudEvent(CloudEventsSpecVersion.Default);
            foreach (var (key, value) in extensions)
            {
                BaseEvent[key] = value;
            }
            BaseEvent.Id = id ?? Guid.NewGuid().ToString();
            BaseEvent.Type = typeof(TData).Name;
            BaseEvent.Source = source;
            BaseEvent.Time = time ?? DateTimeOffset.UtcNow;
            BaseEvent.DataContentType = new ContentType().ToString();

            _applicationNameService = applicationNameService;
            TypedData = data;
        }
```

Reply:
Do you have a resource for this recommendation? I only know about some drawbacks in general with optional parameters: https://rules.sonarsource.com/csharp/RSPEC-2360

But I don't have a strong opinion on this and most usages won't use the optional parameters anyway which gets quite clear with your suggestion as it only leaves the second constructor with two usages which are both inside of this class. (I'm not sure whether users would want to control the time and id directly.) So, this could also allow us to make the second constructor private if we want to.

Reply:
ReSharper and Rider complained about them from time to time but I couldn't find anything from Jetbrains about this very topic right now. 

@farodin91 also noted already that the second constructor could be made `private` considering the snippet above.

In general I tend to introduce multiple constructors for overloads because I consider the resulting IDE completion a bit easier to understand in contrary to a lot of parameters where I do not need to pass all of them but probably a few of them by name or something like that but that's totally personal!

Keep it as you prefer as it was only a nitpick in the first place and I don't have a strong opinion about it :sweat_smile: